src/components/widget-summary.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect} from 'react'
+
+const Summary = () => {
+
+  const [name, setName] = useState(null)
+  const [type, setType] = useState(null)
+  const [icon, setIcon] = useState(null)
+
+  useEffect(() => {
+    waitForWindow().then(() => {
+      setName(window.NAME)
+      setType(window.TYPE)
+      setIcon(window.ICON)
+    })
+  })
+
+  const waitForWindow = async () => {
+		while(!window.hasOwnProperty('NAME')
+		&& !window.hasOwnProperty('TYPE')
+		&& !window.hasOwnProperty('ICON')) {
+			await new Promise(resolve => setTimeout(resolve, 500))
+		}
+	}
+
+  let bodyRender = null
+  if (!!name) {
+    bodyRender = (
+      <div className="widget_info">
+      	<div className="widget_icon">
+      		<img src={ icon } alt=""/>
+      	</div>
+      	<ul className="widget_info">
+          { name && <li className="widget_name">{ name }</li>}
+          { avail && <li className="widget_availability">{ name }</li>}
```

Comment:
`avail` is undefined here - in the original partial view, it's exposed to the view from the `build_widget_login` method of the widgets controller. It'll probably have to be added as a JS global.